### PR TITLE
export stores for more flexibility

### PR DESCRIPTION
**Pull request by thekip**
In some cases, we need to reset internal state of progeny plugin. 
In current version it imposible.
I was ispired by gulp-cached and export internal stores outside. 

Now users of this plugin can clear state whenever he wants. Perhap you may think that this broke incapsulation, yes, i think same. So may be a better way is add method like .reset(), but current solution is much easier and much clearier for the users because they understand what exactly they do. 

This is my case (simplified):

``` js
gulp.task('templates', () => {
const sources = gulp
    .src(paths.src + 'js/views/**/*.jade')
    .pipe(cache('jade-templates'))
    .pipe(progeny({rootPath: globalConfig.paths.projectDir}))
    .pipe(filter(['**/*.jade', '!**/_*.jade']))

return es.merge(_.map(config.langs, (lang) => {
    const strings = i18nCatalog.get(lang);
    const locals = _.extend({}, i18nFuncs(strings));

    return sources
      .pipe(clone())
      .pipe(jade(locals));
      .pipe(gulp.dest(paths.dest + 'j/templates'))
  }));
})

gulp.task('templates-watch', ['templates'], () => {
  const watcher = gulp.watch(paths.src + 'js/views/**/*.jade', ['templates']);
})

gulp.task('translations-watch', ['translations'], () => {
  const watcher = gulp.watch(paths.translates + '*.po', ['templates']); 

  watcher.on('change', () => {
    cache.caches['jade-templates'] = {};
    progeny.processedFileNames = {};
  });
});
```

Yeah, this is simplified :)

We have i18n system builded with gulp. For each language we generate own version of template. 
When templates changed in watch mode (`templates-watch` task) all things do as usual, gulp-cached throw only changed file, progeny add dependecy, all is OK. 

But when changed `.po` file (translations catalog) we have to rebuild all templates. So template sources actually didn't changed  that's why i flush gulp-cache. Therefore ALL templates passed to progeny, but progeny thinks that is not first pass, and for all templates add all dependencies, and it becames a mess. 

image

So if i do

``` js
 progeny.processedFileNames = {};
```

In my `translations-watch` task, all will be work fine. 

In other words this case can be figured as: when locals (template variables) changed we have to reload all templates, even if source code is not changed.

### CODE REVIEW

**Comment by HerringtonDarkholme** [comment]
Why not adding a `reset` method that clear both `processedFileNames` and `depCache`? As you pointed out a `reset` method *feels* more encapsulated. If I understand correctly, passing all files down to progeny will rebuild the whole `depCache` so resetting `processedFileNames` implies resetting `depCache` as well.

Though current approach is fine. Would you please add a test for this pull request? I'm glad to merge your contribution if test is added.

Thank you

**Comment by thekip** [comment]
Unfortunately test fails onmy pc because of Windows. One of first tests is fail because slashes in paths is diffrent.

**Comment by HerringtonDarkholme** [comment]
~~Can you test it on cygwin or bash on windows10~~

I'll fix portability problem shortly. Can you wait it till tomorrow?

**Comment by thekip** [comment]
Yes, it can wait till mondayas well

**Comment by HerringtonDarkholme** [comment]
I have updated some tests. Can you try it again?

**Comment by HerringtonDarkholme** [comment]
ping @thekip
